src/openrct2-ui/windows/NewRide.cpp

```diff
@@ -903,14 +902,12 @@ static void WindowNewRidePaintRideInformation(
     ft.Add<rct_string_id>(rideNaming.Description);
     DrawTextWrapped(dpi, screenPos, width, STR_NEW_RIDE_NAME_AND_DESCRIPTION, ft);
 
-    char availabilityString[AVAILABILITY_STRING_SIZE];
-    WindowNewRideListVehiclesFor(item.Type, rideEntry, availabilityString, sizeof(availabilityString));
+    const auto availabilityString = WindowNewRideListVehiclesFor(item.Type);
```

Comment:
Creating a std::string will call `malloc`. Please avoid calling `malloc` every single frame. An easy way to avoid this, is to store the std::string a field / member of the window. Then you are re-using the same memory each paint.

Reply:
Is this event actually called every frame?

Reply:
Its a paint function so possibly depending on drawing engine.

Reply:
Right, I somehow misread the function name.

Reply:
@IntelOrca Pushed a fix that should hopefully address this.